sky/core.py

```diff
@@ -437,26 +440,29 @@ def autostop(
         operation=operation,
     )
     backend = backend_utils.get_backend_from_handle(handle)
+
+    if tpu_utils.is_tpu_vm_pod(handle.launched_resources):
+        # Reference:
+        # https://cloud.google.com/tpu/docs/managing-tpus-tpu-vm#stopping_a_with_gcloud  # pylint: disable=line-too-long
+        raise exceptions.NotSupportedError(
+            f'{operation} cluster {cluster_name!r} with TPU VM Pod '
+            'is not supported.')
+
     if not isinstance(backend, backends.CloudVmRayBackend):
         raise exceptions.NotSupportedError(
             f'{operation} cluster {cluster_name!r} with backend '
             f'{backend.__class__.__name__!r} is not supported.')
-    elif handle.launched_resources.use_spot and not down and not is_cancel:
+    elif (handle.launched_resources.use_spot and not down and not is_cancel and
+          not handle.launched_resources.cloud.is_same_cloud(clouds.GCP())):
```

Comment:
Similarly, should we make this one of the `CloudImplementationFeatures`?

Reply:
Good call, done.